lib/components/narrative/itinerary-carousel.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component, PropTypes } from 'react'
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import SwipeableViews from 'react-swipeable-views'
+
+import { setActiveItinerary, setActiveLeg, setActiveStep } from '../../actions/narrative'
+import Icon from './icon'
+import NarrativeItinerary from './narrative-itinerary'
+import Loading from './loading'
+import { getActiveSearch } from '../../util/state'
+
+class ItineraryCarousel extends Component {
+  state = {}
+  static propTypes = {
+    itineraries: PropTypes.array,
+    pending: PropTypes.bool,
+    activeItinerary: PropTypes.number,
+    hideHeader: PropTypes.bool,
+    setActiveItinerary: PropTypes.func,
+    setActiveLeg: PropTypes.func,
+    setActiveStep: PropTypes.func
+  }
+  _onItineraryClick = () => {
+    this.setState({expanded: !this.state.expanded})
+  }
+  _leftClick = () => {
+    const newIndex = this.props.activeItinerary === 0 ? this.props.itineraries.length - 1 : this.props.activeItinerary - 1
```

Comment:
If you used destructuring here, it could produce more concise code.  Furthermore, you could return the result of the conditional operation without having to create a new variable.  Same in the _rightClick method.